pkg/tsdb/prometheus/custom_query_params_middleware.go

```diff
@@ -25,21 +24,18 @@ func customQueryParametersMiddleware() sdkhttpclient.Middleware {
 			return next
 		}
 
+		values, err := url.ParseQuery(customQueryParams)
+		if err != nil {
+			log.Error("Failed to parse custom query parameters, skipping middleware", "error", err)
+			return next
+		}
+
 		return sdkhttpclient.RoundTripperFunc(func(req *http.Request) (*http.Response, error) {
-			params := url.Values{}
-			for _, param := range strings.Split(customQueryParams, "&") {
-				parts := strings.Split(param, "=")
-				if len(parts) == 1 {
-					// This is probably a mistake on the users part in defining the params but we don't want to crash.
-					params.Add(parts[0], "")
```

Comment:
when i saw this, i can't help trying to find a test about it. to see whether url.ParseQuery is behaving the same.

Reply:
Keys without values are perfectly allowed in url params, so this is not an error. The only error ParseQuery could ever return is the url decoding error, which previous code didn't check, but should.